pandas/_libs/tslib.pyx

```diff
@@ -239,11 +239,21 @@ def array_with_unit_to_datetime(
         # then need to iterate
         if values.dtype.kind == "i":
             # Note: this condition makes the casting="same_kind" redundant
-            iresult = values.astype('i8', casting='same_kind', copy=False)
+            iresult = values.astype("i8", casting="same_kind", copy=False)
             # fill by comparing to NPY_NAT constant
             mask = iresult == NPY_NAT
             iresult[mask] = 0
-            fvalues = iresult.astype('f8') * m
+            fvalues = iresult.astype("f8") * m
+            need_to_iterate = False
+
+        # GH20445
+        elif values.dtype.kind == "f":
+            # Note: this condition makes the casting="same_kind" redundant
+            fresult = values.astype("f8", casting="same_kind", copy=False)
+            # fill by comparing to NPY_NAT constant
+            mask = fresult == NPY_NAT
+            fresult[mask] = 0
+            fvalues = fresult * (<float64_t>m)
```

Comment:
This cast doesn't seem right and is probably the cause of your portability issue. What is the warning that this generates without?

Reply:
The traceback if I remove this line:
```
  File "/workspaces/pandas-arw2019/pandas/core/tools/datetimes.py", line 802, in to_datetime
    values = convert_listlike(arg._values, format)
  File "/workspaces/pandas-arw2019/pandas/core/tools/datetimes.py", line 338, in _convert_listlike_datetimes
    result, tz_parsed = tslib.array_with_unit_to_datetime(
  File "pandas/_libs/tslib.pyx", line 262, in pandas._libs.tslib.array_with_unit_to_datetime
    if ((fvalues < Timestamp.min.value).any()
SystemError: /home/conda/feedstock_root/build_artifacts/python_1591030388223/work/Objects/object.c:769: bad argument to internal function
```